generators/argonaut-r4/metadata_extractor.rb

```diff
@@ -119,6 +119,39 @@ def add_interactions(resource, sequence)
     end
   end
 
+  def add_must_support_elements(profile_definition, sequence)
+    profile_definition['snapshot']['element'].select { |el| el['mustSupport'] }.each do |element|
+      if element['path'].end_with? 'extension'
+        sequence[:must_supports] <<
+          {
+            type: 'extension',
+            id: element['id'],
+            path: element['path'],
+            url: element['type'].first['profile'].first
+          }
+        next
+      end
+
+      path = element['path']
+      if path.include? '[x]'
+        variable_el = profile_definition['snapshot']['element'].select { |el| el['id'] == (path.split('[x]').first + '[x]') }.first
```

Comment:
Use `.find {}` instead of `.select {}.first`. Also, "choice elements" is the official term for these, so maybe `variable` should be replaced with `choice`.

Reply:
Changed to find and used choice.